lib/curator.rb

```diff
@@ -0,0 +1,56 @@
+class Curator
+  attr_reader :photographs, :artists
+
+  def initialize
+    @photographs = []
+    @artists = []
+  end
+
+
+  def add_photograph(photo)
+    @photographs << photo
+  end
+
+  def add_artist(artist)
+    @artists << artist
+  end
+
+  def find_artist_by_id(id)
+    found = []
+    @artists.each do |artist|
+      found << artist if artist.id == id
+      end
+    @found
+  end
+
+  def find_photograph_by_id(id)
+    found = []
+    @photographs.each do |photo|
+      found << photo if photo.id == id
+    end
+    found
+  end
```

Comment:
These two methods, if you look at the interaction patterns, were expecting single objects to come back; when we search for all artists named "John Smith" we might find several, but ID values are meant to be unique and we'd only want to get one of those. I think your implementation here for these last two methods probably made other work more difficult than it needed to be.